Core/src/org/sleuthkit/autopsy/healthmonitor/ServicesHealthMonitor.java

```diff
@@ -0,0 +1,767 @@
+     * @throws HealthMonitorException 
+     */
+    static synchronized void startUp() throws HealthMonitorException {
+        getInstance();
+    }
+    
+    /**
+     * Enabled/disable the health monitor.
+     * @param enabled true to enable the monitor, false to disable it
+     * @throws HealthMonitorException 
+     */
+    static synchronized void setEnabled(boolean enabled) throws HealthMonitorException {
+        if(enabled == isEnabled.get()) {
+            // The setting has not changed, so do nothing
+            return;
+        }
+        
+        if(enabled) {
+            getInstance().activateMonitor();
+            
+            // If activateMonitor fails, we won't update either of these
+            ModuleSettings.setConfigSetting(MODULE_NAME, IS_ENABLED_KEY, "true");
+            isEnabled.set(true);
+        } else {
+            ModuleSettings.setConfigSetting(MODULE_NAME, IS_ENABLED_KEY, "false");
+            isEnabled.set(false);
+            getInstance().deactivateMonitor();
+        }
+    }
+    
+    /**
+     * Get a metric that will measure the time to execute a section of code.
+     * Call this before the section of code to be timed and then
+     * submit it afterward using submitTimingMetric().
+     * This method is safe to call regardless of whether the Services Health
+     * Monitor is enabled.
+     * @param name A short but descriptive name describing the code being timed.
+     *             This name will appear in the UI.
+     * @return The TimingMetric object
+     */
+    public static TimingMetric getTimingMetric(String name) {
+        if(isEnabled.get()) {
+            return new TimingMetric(name);
+        }
+        return null;
+    }
+    
+    /**
+     * Submit the metric that was previously obtained through getTimingMetric().
+     * Call this immediately after the section of code being timed.
+     * This method is safe to call regardless of whether the Services Health
+     * Monitor is enabled.
+     * @param metric The TimingMetric object obtained from getTimingMetric()
+     */
+    public static void submitTimingMetric(TimingMetric metric) {
+        if(isEnabled.get() && (metric != null)) {
+            metric.stopTiming();
+            try {
+                getInstance().addTimingMetric(metric);
+            } catch (HealthMonitorException ex) {
+                // We don't want calling methods to have to check for exceptions, so just log it
+                logger.log(Level.SEVERE, "Error adding timing metric", ex);
+            }
+        }
+    }
+    
+    /**
+     * Add the timing metric data to the map.
+     * @param metric The metric to add. stopTiming() should already have been called.
+     */
+    private void addTimingMetric(TimingMetric metric) throws HealthMonitorException {
+
+        // Do as little as possible within the synchronized block to minimize
+        // blocking with multiple threads.
+        synchronized(this) {
```

Comment:
Could this synchronized block be eliminated if timingInfoMap was a ConcurrentHashMap?